src/foremast/utils/dns.py

```diff
@@ -91,6 +92,30 @@ def update_dns_zone_record(env, zone_id, **kwargs):
 
     LOG.debug('Route53 JSON Response: \n%s', pformat(response))
 
+def find_existing_record(env, zone_id, dns_name, valuecheck=[]):
```

Comment:
Can you split `valuecheck` into `check_type` and `check_value` keyword arguments instead? I'm not fond of passing 2-tuples around.